lib/ui/painting/immutable_buffer.cc

```diff
@@ -101,20 +143,28 @@ Dart_Handle ImmutableBuffer::initFromFile(Dart_Handle raw_buffer_handle,
 
   auto* dart_state = UIDartState::Current();
   auto ui_task_runner = dart_state->GetTaskRunners().GetUITaskRunner();
-  auto buffer_callback =
-      std::make_unique<tonic::DartPersistentValue>(dart_state, callback_handle);
-  auto buffer_handle = std::make_unique<tonic::DartPersistentValue>(
-      dart_state, raw_buffer_handle);
+  auto* buffer_callback_ptr =
+      new tonic::DartPersistentValue(dart_state, callback_handle);
+  auto* buffer_handle_ptr =
+      new tonic::DartPersistentValue(dart_state, raw_buffer_handle);
 
   auto ui_task = fml::MakeCopyable(
-      [buffer_callback = std::move(buffer_callback),
-       buffer_handle = std::move(buffer_handle)](const sk_sp<SkData>& sk_data,
-                                                 size_t buffer_size) mutable {
-        auto dart_state = buffer_callback->dart_state().lock();
+      [buffer_callback_ptr, buffer_handle_ptr](const sk_sp<SkData>& sk_data,
+                                               size_t buffer_size) mutable {
+        if (!buffer_callback_ptr) {
+          return;
+        }
+        auto dart_state = buffer_callback_ptr->dart_state().lock();
         if (!dart_state) {
           return;
         }
         tonic::DartState::Scope scope(dart_state);
+
+        std::unique_ptr<tonic::DartPersistentValue> buffer_handle(
```

Comment:
Looked into this, and it's actually ok to destruct the `tonic::DartPersistentValue` outside of a `DartState::Scope`.  The `DartPersistentValue` holds a weak reference to the `DartState` used during construction and will enter that `DartState`'s scope to delete the wrapped value.

So these `unique_ptr`s should be moved to the top of the function in order to ensure that they are not leaked if it returns early.